src/Microsoft.DocAsCode.Common/Git/GitUtility.cs

```diff
@@ -59,7 +59,31 @@ public static GitRepoInfo GetRepoInfo(string directory)
                 throw new GitException("Can't find git command in current environment");
             }
 
-            return Cache.GetOrAdd(directory, GetRepoInfoCore);
+            if (IsGitRoot(directory))
+            {
+                return Cache.GetOrAdd(directory, GetRepoInfoCore);
+            }
+
+            var parentDirInfo = Directory.GetParent(directory);
+            return Cache.GetOrAdd(directory, GetRepoInfo(parentDirInfo?.FullName));
```

Comment:
why are there more calling ExistGitCommand()? I think it's the same as before.  However, I should reduce the number of times to call ExistGitCommand too...

---
In reply to: [94094370](https://github.com/dotnet/docfx/pull/1077#discussion_r94094370) [](ancestors = 94094370)